### DISCUSSION TITLE
Is it possible to create an action to do CRTDSPF or CRTPRTF

### DISCUSSION THREAD

**Discussion by PatRose** [codefori/vscode-ibmi#1737] [Q&A]
We are working with transitioning to the VS Code structure, but are trying to determine what the best practice would be to compile DSP and PRT files within this IBMi structure.

From what we can tell, we will need to manually transfer the IFS file (or local file) to the Object and then run CRTDSPF or CRTPRTF. Is there a simple way we can automate these two steps so that it is consistent for all developers and allows us to remain in the VSC environment when doing development?

**Comment by chrjorgensen** [comment]
Indeed, you can!

You can have all your source in the IFS - even those source types, where the CRT-command does not support IFS - and compile from the streamfiles by using the excellent command [CRTFRMSTMF](https://github.com/BrianGarland/CRTFRMSTMF) by @BrianGarland. I've been using this command for years with no problems at all. It will copy the streamfile to a temporary source member and forward the parameters to the CRT-command. You can incorporate the CRTFRMSTMF command in your actions, make-file or however you build your objects.

This will allow you to have ALL source in the IFS, thus in a git repository tracking all changes to the source.
